astropy/units/tests/test_format.py

```diff
@@ -1110,3 +1110,15 @@ def test_Fits_name_deprecation():
     ):
         from astropy.units.format import Fits
     assert Fits is u.format.FITS
+
+
+def test_to_string_unknown_format():
+    # Check full message to ensure we correctly classify in-out and
```

Comment:
You've already written a test for this: https://github.com/astropy/astropy/blob/28d1279744521d5d98e56b76f7f1f0259e2bcc41/astropy/units/tests/test_format.py#L1034-L1036

Reply:
OK, clearly I should not just have looked for tests with `.to_string()`...  I now combined the two tests in a new class, with both checking the full message.